src/ReportManager/SchemaFormContext/SchemaForm/useFieldDetails/index.js

```diff
@@ -0,0 +1,15 @@
+import { useContext } from 'react';
+
+import { SchemaFormContext } from '../SchemaFormContext';
+
+const useFieldDetails = (fieldName) => {
+  const { getSectionDetails, getFieldDetails, getHeaderDetails, isSection, isField } = useContext(SchemaFormContext);
```

Comment:
Again, some of these functions should be utilities, not context passed methods. Context should not be used as a way to pass random stuff, but specifically state related variables that are expected to cause re-renders.

Reply:
Already changed it 🫡